### PR TITLE
79242 api add created at and by fields

### PR DESCRIPTION
**Pull request by hbjorgo**
https://dev.azure.com/statoildeveloper/Statoil.NextGenProCoSys/_workitems/edit/79242

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.Query/GetAttachments/GetAttachmentsQueryHandler.cs [hunk 1]
```diff
@@ -30,12 +41,19 @@ public async Task<Result<List<AttachmentDto>>> Handle(GetAttachmentsQuery reques
                 return new NotFoundResult<List<AttachmentDto>>($"Invitation with ID {request.InvitationId} not found");
             }
 
+            var uploadedByIds = invitation.Attachments.Select(a => a.UploadedById).ToList();
+            var uploadedBys = await _context.QuerySet<Person>().Where(x => uploadedByIds.Contains(x.Id)).ToListAsync();
```
Comment (llykkelarsen) [inline src/Equinor.ProCoSys.IPO.Query/GetAttachments/GetAttachmentsQueryHandler.cs:45]: pass cancellationtoken to tolist method

File: src/Equinor.ProCoSys.IPO.Query/GetAttachments/GetAttachmentsQueryHandler.cs [hunk 2]
```diff
@@ -30,12 +41,19 @@ public async Task<Result<List<AttachmentDto>>> Handle(GetAttachmentsQuery reques
...
+            var uploadedByDtos = uploadedBys.Select(x => new PersonDto(x.Id, x.FirstName, x.LastName, x.Oid, null, x.RowVersion.ConvertToString())).ToDictionary(x => x.Id);
```
Comment (llykkelarsen) [inline src/Equinor.ProCoSys.IPO.Query/GetAttachments/GetAttachmentsQueryHandler.cs:46]: long line

File: src/Equinor.ProCoSys.IPO.WebApi/Controllers/Invitation/InvitationsController.cs [hunk 3]
```diff
@@ -117,7 +117,7 @@ public class InvitationsController : ControllerBase
             return this.FromResult(result);
         }
 
-        [Authorize(Roles = Permissions.IPO_READ)]
+        //[Authorize(Roles = Permissions.IPO_READ)]
```
Comment (llykkelarsen) [inline src/Equinor.ProCoSys.IPO.WebApi/Controllers/Invitation/InvitationsController.cs:120]: undo

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachmentById/AttachmentExtensionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+using Equinor.ProCoSys.IPO.Test.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Equinor.ProCoSys.IPO.Query.Tests.GetAttachmentById
+{
+    [TestClass]
+    public class AttachmentExtensionsTests
+    {
+
```
Comment (llykkelarsen) [inline src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachmentById/AttachmentExtensionsTests.cs:16]: whitespace

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachmentById/AttachmentExtensionsTests.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Globalization;
```
Comment (llykkelarsen) [inline src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachmentById/AttachmentExtensionsTests.cs:2]: remove

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachments/GetAttachmentsQueryHandlerTests.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Query.Tests.GetAttachments
+{
+    [TestClass]
+    public class GetAttachmentsQueryHandlerTests : ReadOnlyTestsBase
+    {
+        private Mock<IOptionsMonitor<BlobStorageOptions>> blobStorageOptionsMonitorMock;
```
Comment (llykkelarsen) [inline src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachments/GetAttachmentsQueryHandlerTests.cs:19]: _blobStorageOptionsMonitorMock

**Review by llykkelarsen — APPROVED** [review]

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachmentById/GetAttachmentByIdQueryHandlerTests.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Query.Tests.GetAttachmentById
+{
+    [TestClass]
+    public class GetAttachmentsQueryHandlerTests : ReadOnlyTestsBase
+    {
+        private Mock<IOptionsMonitor<BlobStorageOptions>> blobStorageOptionsMonitorMock;
```
Comment (llykkelarsen) [inline src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetAttachmentById/GetAttachmentByIdQueryHandlerTests.cs:18]: _blobStorageOptionsMonitorMock

File: src/Equinor.ProCoSys.IPO.WebApi/Controllers/Invitation/InvitationsController.cs (hunk 3 above)
Comment (hbjorgo) [reply src/Equinor.ProCoSys.IPO.WebApi/Controllers/Invitation/InvitationsController.cs]: :o

**Pull request merged**